WordPressAuthenticator/GoogleSignIn/String+Random.swift

```diff
@@ -0,0 +1,22 @@
+extension String {
+
+    static func randomString(using characters: Set<Character>, withLength length: Int) -> String {
+        let allowedCharactersCount = UInt32(characters.count)
+
+        var randomBytes = [UInt8](repeating: 0, count: length)
+        // Use `SecRandomCopyBytes` to generate cryptographically secure random bytes to use as
+        // offsets to create the random string.
+        let status = SecRandomCopyBytes(kSecRandomDefault, length, &randomBytes)
+
+        // FIXME: Handle errors by returning nil
+        guard status == errSecSuccess else {
+            fatalError()
+        }
```

Comment:
The `SecRandomCopyBytes` function can fail for a variety of reasons, including:

<details>
<summary>See some way it can fail.</summary>
Insufficient entropy: The function may fail if there is not enough entropy available to generate the requested number of random bytes. This can happen if the system's random number generator has not collected enough entropy from hardware events or other sources. In this case, the function may return a errSecInsufficientEntropy error code.

Invalid input parameters: The function may fail if the input parameters are invalid or out of range. For example, if the output buffer pointer is null, the function will return a errSecParam error code.

Security framework errors: The function may fail due to errors within the Security framework itself. For example, if the Security framework is not initialized properly or encounters an internal error, the function may return a errSecUnimplemented error code.

System errors: The function may fail due to system-level errors, such as memory allocation failures or I/O errors. In this case, the function may return a errSecIO error code.

To handle possible errors, you should always check the return value of the `SecRandomCopyBytes` function and handle any errors appropriately. If the function returns an error code, you can use the `SecCopyErrorMessageString` function to get a human-readable error message that can help you diagnose the problem.

</details>

(_via ChatGPT_).

I haven't handled returning `nil` from here yet because of the downstream usages.

But, I think we can handle this without propagating the error outside the library by falling back to a less secure but runtime-guaranteed random value in case of failure, which I think it's reasonable to assume would be incredibly rare IRL.

Reply:
Yeah, I'm not sure I've _ever_ seen `SecRandomCopyBytes` fail in real life, so if it did I'd take that as a sign that something else on the system is catastrophically wrong – my suggestion there would be to `throw` all the way back up to the UI and tell the user "we couldn't log you in, please try again" or whatever

Reply:
Fair enough. We'll need to do _some_ error handling at the UI level anyway, so is not like adding a specific error for this one comes at a high cost.

Reply:
See 60181fa0